internal/controller/ngrok/ngroktrafficpolicy_controller.go

```diff
@@ -63,7 +66,25 @@ type NgrokTrafficPolicyReconciler struct {
 func (r *NgrokTrafficPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
 
-	err := r.Driver.SyncEdges(ctx, r.Client)
+	policy := &ngrokv1alpha1.NgrokTrafficPolicy{}
+	if err := r.Get(ctx, req.NamespacedName, policy); err != nil {
+		return ctrl.Result{}, err
+	}
+
+	parsedTrafficPolicy, err := util.NewTrafficPolicyFromJson(policy.Spec.Policy)
+	if err != nil {
+		return ctrl.Result{}, err
```

Comment:
We might also want to record an event here if we can't parse the policy for whatever reason. Probably also in the other places we unmarshal the policy from json as well.